components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java

```diff
@@ -999,26 +1000,26 @@ private void validateTemplateType(String templateDisplayName, String tenantDomai
         List<EmailTemplate> templateList = new ArrayList<>();
         Collection templateType = (Collection) resourceMgtService.getIdentityResource(templateTypeRegistryPath,
                 tenantDomain);
-        if (templateType != null) {
-            for (String template : templateType.getChildren()) {
-                Resource templateResource = resourceMgtService.getIdentityResource(template, tenantDomain);
-                if (templateResource != null) {
-                    try {
-                        EmailTemplate templateDTO = I18nEmailUtil.getEmailTemplate(templateResource);
-                        templateList.add(templateDTO);
-                    } catch (I18nEmailMgtException ex) {
-                        log.error(ex.getMessage(), ex);
-                    }
-                }
-            }
-            return templateList;
-        } else {
+
+        if (templateType == null) {
             String type = templateTypeRegistryPath.split(PATH_SEPARATOR)[
                     templateTypeRegistryPath.split(PATH_SEPARATOR).length - 1];
             String message =
                     String.format("Email Template Type: %s not found in %s tenant registry.", type, tenantDomain);
-            throw new I18nEmailMgtClientException(message);
+            throw new I18nEmailMgtClientException(EMAIL_TEMPLATE_TYPE_NOT_FOUND, message);
         }
+        for (String template : templateType.getChildren()) {
+            Resource templateResource = resourceMgtService.getIdentityResource(template, tenantDomain);
+            if (templateResource != null) {
+                try {
+                    EmailTemplate templateDTO = I18nEmailUtil.getEmailTemplate(templateResource);
+                    templateList.add(templateDTO);
+                } catch (I18nEmailMgtException ex) {
+                    log.error(ex.getMessage(), ex);
```

Comment:
ex.getMessage() is anyway getting logged when we log ex.. therefore shall we add a descriptive error message here?

Reply:
addressed in 52e7a7a13a22dbfc30b82329ce3be5de9d523b6a